File: esports-backend/src/main/java/com/esports/dto/RegisterRequest.java
package com.esports.dto;

import com.esports.entity.User;
import com.esports.validation.PasswordMatches;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@PasswordMatches
public class RegisterRequest {
    @NotBlank(message = "Name is required")
    @Pattern(regexp = "^[a-zA-Z]+(\\s[a-zA-Z]+)?$", message = "Name must contain only alphabets and at most one space")
    private String name;

    @NotBlank(message = "Email is required")
    @Pattern(
    	    regexp = "^[a-zA-Z0-9._%+-]+@(gmail\\.com|admin\\.com)$",
    	    message = "Email must be from gmail.com domain"
    	)
    private String email;

    @NotBlank(message = "Password is required")
    @Pattern(
    	    regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[!@#$%^&*()_+\\-=\\[\\]{};':\"\\\\|,.<>\\/?]).{6,}$",
    	    message = "Password must contain at least one uppercase letter, one lowercase letter, one number, and one special character"
    	)
    private String password;

    @NotBlank(message = "Confirm password is required")
    private String confirmPassword;

    @NotBlank(message = "Role is required")
    @Pattern(regexp = "^(CREATOR|FREELANCER)$", message = "Role must be either CREATOR or FREELANCER")
    private String role;
}

File: esports-backend-dotnet/Upkart1/DTO/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Upkart1.Validation;

namespace Upkart1.DTO
{
    [PasswordMatches]
    public class RegisterRequest
    {
        [Required(ErrorMessage = "Name is required")]
        [RegularExpression(@"^[a-zA-Z]+(\s[a-zA-Z]+)?$", ErrorMessage = "Name must contain only alphabets and at most one space")]
        [JsonPropertyName("name")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Email is required")]
        [RegularExpression(@"^[a-zA-Z0-9._%+-]+@(gmail\.com|admin\.com)$", ErrorMessage = "Email must be from gmail.com domain")]
        [JsonPropertyName("email")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Password is required")]
        [RegularExpression(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*()_+\-=\[\]{};':""\\|,.<>\/?]).{6,}$", ErrorMessage = "Password must contain at least one uppercase letter, one lowercase letter, one number, and one special character")]
        [JsonPropertyName("password")]
        public string Password { get; set; }

        [Required(ErrorMessage = "Confirm password is required")]
        [JsonPropertyName("confirmPassword")]
        public string ConfirmPassword { get; set; }

        [Required(ErrorMessage = "Role is required")]
        [RegularExpression(@"^(CREATOR|FREELANCER)$", ErrorMessage = "Role must be either CREATOR or FREELANCER")]
        [JsonPropertyName("role")]
        public string Role { get; set; }
    }
}
